app/src/main/java/com/concordium/wallet/ui/walletconnect/WalletConnectViewModel.kt

```diff
@@ -1364,31 +1404,79 @@ private constructor(
         return sessionRequestIdentityProofIdentities[account.identityId]
     }
 
-    private suspend fun signRequestedMessage(accountKeys: AccountData) {
-        val signMessageInput = SignMessageInput(
-            address = sessionRequestAccount.address,
-            message = sessionRequestMessageToSign,
-            keys = accountKeys,
-        )
-        val signMessageOutput = App.appCore.cryptoLibrary.signMessage(signMessageInput)
-        if (signMessageOutput == null) {
-            Log.e("failed_signing_message")
+    private fun signRequestedMessage(accountKeys: AccountData) {
+        val signatureHex = try {
+            val signKeyHex =
+                App.appCore.gson.fromJson(accountKeys.keys.json, AccountDataKeys::class.java)
+                    .level0
+                    .keys
+                    .keys
+                    .signKey
+
+            val message = when (val signMessageParams = sessionRequestSignMessageParams) {
+                is SignMessageParams.Binary ->
+                    signMessageParams.data.hexToBytes()
+
+                is SignMessageParams.Text ->
+                    signMessageParams.data.encodeToByteArray()
+            }
+
+            // See https://github.com/Concordium/concordium-base/blob/main/mobile_wallet/src/lib.rs#L89
+            val input = SHA256.hash(
+                AccountAddress.from(sessionRequestAccount.address).bytes
+                        + ByteArray(8)
+                        + message
+            )
+
+            ED25519SecretKey.from(signKeyHex)
+                .sign(input)
+                .toHex()
+        } catch (e: Exception) {
+            Log.e("failed_signing_message", e)
+
             respondError(
-                message = "Failed signing message: crypto library internal error"
+                message = "Failed signing message: $e"
             )
 
             mutableEventsFlow.tryEmit(
                 Event.ShowFloatingError(
                     Error.CryptographyFailed
                 )
             )
-        } else {
-            respondSuccess(
-                result = Gson().toJson(signMessageOutput)
-            )
 
-            mutableStateFlow.tryEmit(State.Idle)
-            handleNextOldestPendingSessionRequest()
+            return
+        }
+
+        respondSuccess(
+            result = App.appCore.gson.toJson(SignMessageOutput(X0(signatureHex)))
+        )
+
+        mutableStateFlow.tryEmit(State.Idle)
+        handleNextOldestPendingSessionRequest()
+    }
+
+    fun onShowSignRequestDetailsClicked() = viewModelScope.launch {
+        val reviewState = state as? State.SessionRequestReview.SignRequestReview
+        check(reviewState != null && reviewState.canShowDetails) {
+            "Show details button can only be clicked in the sign request review state " +
+                    "allowing showing the details"
+        }
```

Comment:
Is the Kotlin linter breaking the string in that way? If not, I suggest:

```suggestion
        check(reviewState != null && reviewState.canShowDetails) {
            "Show details button can only be clicked in the sign request review state 
                    allowing showing the details"
        }
```